fsharp-backend/src/ApiServer/Api/APISecrets.fs

```diff
@@ -92,3 +92,73 @@ module Delete =
                 { secret_name = s.name; secret_value = s.value })
               secrets }
     }
+
+module InsertV1 =
+
+  type Secret = { name : string; value : string }
+
+  type Params = Secret
+
+  type T = { secrets : List<Secret> }
+
+  /// API endpoint to insert a Secret within a canvas
+  let insert (ctx : HttpContext) : Task<T> =
+    task {
+      use t = startTimer "read-api" ctx
+      try
+        t.next "read-api"
+        let canvasInfo = loadCanvasInfo ctx
+        let! p = ctx.ReadVanillaJsonAsync<Params>()
+        Telemetry.addTags [ "secretName", p.name ]
+
+        t.next "insert-secret"
+        do! LibBackend.Secret.insert canvasInfo.id p.name p.value
+
+        t.next "get-secrets"
+        let! secrets = LibBackend.Secret.getCanvasSecrets canvasInfo.id
+
+        t.next "write-api"
+        return
+          { secrets =
+              List.map
+                (fun (s : PT.Secret.T) -> { name = s.name; value = s.value })
+                secrets }
```

Comment:
Not something we need to cover in this PR, but:
I think it'd be reasonable for the client to fetch the secrets list manually after getting an OK response back here

Reply:
Yeah, the APIs could do with some work, certainly.